lib/usb_connection.js

```diff
@@ -175,12 +178,18 @@ USB.Connection.prototype._close = function(callback) {
   }
 
   self.closed = true;
-  self.intf.release(true, function() {
-    self.device.close();
-    if (typeof callback === 'function') {
-      callback();
-    }
-  }.bind(self));
+  self.epIn.stopPoll(function() {
+    self.intf.release(true, function attempt() {
+      try {
+        self.device.close();
+      } catch (e) {
+        setTimeout(attempt, 100);
```

Comment:
It would be great if we had a timeout limit to reduce the chance of an infinite loop.